scipy/stats/_stats_py.py

```diff
@@ -9740,9 +9805,58 @@ def wasserstein_distance(u_values, v_values, u_weights=None, v_weights=None):
     >>> wasserstein_distance([3.4, 3.9, 7.5, 7.8], [4.5, 1.4],
     ...                      [1.4, 0.9, 3.1, 7.2], [3.2, 3.5])
     4.0781331438047861
-
+    >>> wasserstein_distance([[0, 2, 3], [1, 2, 5]], [[3, 2, 3], [4, 2, 5]])
+    3.0
+    >>> wasserstein_distance([[0, 2.75], [2, 209.3], [0, 0]],
+    ...                      [[0.2, 0.322], [4.5, 25.1808]],
+    ...                      [0.4, 5.2, 0.114], [0.8, 1.5])
+    174.15840245217169
     """
-    return _cdf_distance(1, u_values, v_values, u_weights, v_weights)
+    u_values, u_weights = _validate_distribution(u_values, u_weights)
+    v_values, v_weights = _validate_distribution(v_values, v_weights)
+
+    # convert value into 2D array
+    m, n = len(u_values), len(v_values)
+    u_values = asarray(u_values).reshape(m, -1)
+    v_values = asarray(v_values).reshape(n, -1)
+
+    # if dimensions are not equal throw error
+    if u_values.shape[1] != v_values.shape[1]:
+        raise ValueError('Dimensions of value are not equal.')
+
+    # if data is 1D then call the cdf_distance function
+    if u_values.shape[1] == 1 and v_values.shape[1] == 1:
+        return _cdf_distance(1,
+                             u_values.flatten(), v_values.flatten(),
+                             u_weights, v_weights)
+
+    # if data contains np.inf then return inf or nan
+    if np.any(np.isinf(u_values)) ^ np.any(np.isinf(v_values)):
+        return np.inf
+    elif np.any(np.isinf(u_values)) and np.any(np.isinf(v_values)):
+        return np.nan
+
+    # create constraints
+    _row_func = np.vectorize(lambda x: np.floor(x/n) if x < m*n else m + x % n)
+    row = _row_func(np.arange(m * n * 2))
+    col = np.append(np.arange(m * n), np.arange(m * n))
+    value = np.ones(m * n * 2, dtype=float)
+    # sparse constraint matrix of size (m + n)*(m * n)
+    A = coo_matrix((value, (row, col)), shape=(n + m, m * n))
+
+    # get cost matrix
+    D = distance_matrix(u_values, v_values, p=2)
+    cost = D.reshape((n * m))
```

Comment:
The linter is flagging this as "extraneous parenthesis"

Reply:
Hi melissa, yes the linter is correct on this, single brackets `( n * m )` works fine with `reshape` and is equivalent with the current double brackets way. I'm doing this simply because Matt suggested [here](https://github.com/scipy/scipy/pull/17473#discussion_r1132655107).